packages/server-wallet/src/__test__/sync-objectives.test.ts

```diff
@@ -0,0 +1,155 @@
+  ]);
+  a = await Wallet.create(aWalletConfig);
+  b = await Wallet.create(bWalletConfig);
+
+  await seedAlicesSigningWallet(a.knex);
+  await seedBobsSigningWallet(b.knex);
+});
+afterAll(async () => {
+  await Promise.all([a.destroy(), b.destroy()]);
+  await Promise.all([DBAdmin.dropDatabase(aWalletConfig), DBAdmin.dropDatabase(bWalletConfig)]);
+});
+
+test('Objectives can be synced if a message is lost', async () => {
+  const createChannelParams: CreateChannelParams = createChannelArgs();
+
+  // We mimic not receiving a message containing objectives
+  const messageToLose = await a.createChannel(createChannelParams);
+
+  const channelId = messageToLose.channelResults[0].channelId;
+  const objectiveId = `OpenChannel-${channelId}`;
+
+  // Only A should have the objective since we "lost" the message
+  expect(await getObjective(a.knex, objectiveId)).toBeDefined();
+  expect(await getObjective(b.knex, objectiveId)).toBeUndefined();
+
+  // We would then call sync after some time of waiting and not making progress
+  const syncResult = await a.syncObjectives([objectiveId]);
+
+  // We should not see the objective yet
+  expect(await getObjective(b.knex, objectiveId)).toBeUndefined();
+
+  // After sync funding should continue as normal
+  const resultB0 = await b.pushMessage(getPayloadFor(bob().participantId, syncResult.outbox));
+  expect(getChannelResultFor(channelId, resultB0.channelResults)).toMatchObject({
+    status: 'proposed',
+    turnNum: 0,
+  });
+});
+
+test('handles the objective being synced even if no message is lost', async () => {
+  const createChannelParams: CreateChannelParams = createChannelArgs();
+
+  // We mimic not receiving a message containing objectives
+  const messageResponse = await a.createChannel(createChannelParams);
+
+  const channelId = messageResponse.channelResults[0].channelId;
+  const objectiveId = `OpenChannel-${channelId}`;
+
+  // The initial message is received
+  await b.pushMessage(getPayloadFor(bob().participantId, messageResponse.outbox));
+
+  // We expect both objectives to be there
+  expect(await getObjective(a.knex, objectiveId)).toBeDefined();
+  expect(await getObjective(b.knex, objectiveId)).toBeDefined();
+
+  const syncResult = await a.syncObjectives([objectiveId]);
+  // Now we push in the sync payload
+  const {outbox, newObjectives, channelResults} = await b.pushMessage(
+    getPayloadFor(bob().participantId, syncResult.outbox)
+  );
+
+  // The only expected result is a sync channel response
+  expect(outbox).toHaveLength(1);
+  expect(outbox[0]).toMatchObject({
+    method: 'MessageQueued',
+    params: {
+      recipient: 'alice',
+      sender: 'bob',
+      data: {
+        signedStates: [expect.objectContaining({turnNum: 0})],
+      },
+    },
+  });
+  expect(newObjectives).toHaveLength(0);
+  expect(channelResults).toHaveLength(1);
+});
+
+test('Can successfully push the sync objective message multiple times', async () => {
```

Comment:
This test describes pushing the sync objective message multiple times, but it seems like it was only pushed once. Am I missing something?

Reply:
We push on line 129 and line 131 but I could add comments making that clearer.

Reply:
yes, my apologies, I missed that.